src/middlewared/middlewared/plugins/smb.py

```diff
@@ -293,22 +293,18 @@ async def netif_wait(self, timeout=120):
 
     @private
     @job(lock="smb_configure")
-    async def configure(self, job, create_paths=True):
+    async def configure(self, job):
         """
         Many samba-related tools will fail if they are unable to initialize
         a messaging context, which will happen if the samba-related directories
         do not exist or have incorrect permissions.
         """
         job.set_progress(0, 'Setting up SMB directories.')
-        if create_paths:
-            await self.setup_directories()
+        await self.setup_directories()
 
         job.set_progress(10, 'Generating stub SMB config.')
         await self.middleware.call('etc.generate', 'smb')
 
-        job.set_progress(25, 'generating SMB, idmap, and directory service config.')
-        await self.middleware.call('etc.generate', 'smb')
-
```

Comment:
`setup_directories` calls `etc.generate 'smb'`

Now that we **always** call `setup_directories` here do we still need to call it again here?

Reply:
Good eye.